core/src/main/java/com/linecorp/armeria/common/AbstractRequestContextBuilder.java

```diff
@@ -433,6 +436,24 @@ protected final String path() {
         return path;
     }
 
+    /**
+     * Sets the UUID of the request. If not set, it is auto-generated with UUID
+     */
+    public AbstractRequestContextBuilder uuid(UUID uuid) {
+        this.uuid = requireNonNull(uuid, "uuid");
+        return this;
+    }
+
+    /**
+     * Returns the {@link UUID} of the request.
+     */
+    protected final UUID uuid() {
+        if(uuid == null) {
+            uuid = UUID.randomUUID();
+        }
```

Comment:
We don't need this because it's always not null?

Reply:
Correct. `uuid` must start with `null`.